### ISSUE TITLE
LineString crashes when zooming out with SKGLSurfaceView on Android

### ISSUE THREAD

**Issue by pauldendulk** [Mapsui/Mapsui#862]
To reproduce, 
- Check out branch https://github.com/Mapsui/Mapsui/tree/research/linestring-hang-in-gl.
- Start Mapsui.Samples.Droid.
- Select the 'Many Vertices' sample under 'Special'.
- Pan a bit to see it behaves fair. A bit sluggish bug okay.
- Zoom out and it becomes more sluggish.
- Zoom further out and the map becomes unresponsive.

The same behaves normally when using the SKCanvasView instead of the SKGLSurfaceView.

**Comment by pauldendulk** [comment]
@charlenni I am going to revert iOS and Android to non GL versions. 😔

We need to submit this at skiasharp.

**Comment by charlenni** [comment]
@pauldendulk Do you really think, this is the best way? You will loose the speed of the GPU backend (see [here](https://github.com/Mapsui/Mapsui/issues/676#issuecomment-589864797)) for a use case, that happens one time in a year. And only on Android, because if you use the same sample on UWP, than there are no problems.

I would suggest to leave it as it is and create an issue at SkiaSharp or direct in Skia.

**Comment by pauldendulk** [comment]
Better performance is seen immediately. Instant gratification, so very attractive. But in this case the downside may be obscure crashes that are sometimes not easy to reproduce and show only up in user logs. This can be a nightmare in a later phase of development where a customer demands stability before they accept the release. 

Stuff like [this](https://github.com/Mapsui/Mapsui/issues/852) and [this](https://github.com/mono/SkiaSharp/issues/1168) could be related to GL instability. Crashes like this are rarely just a simple bug. It usually means they don' t know how to properly management the native resources. This has been a problem for years now, I don't expect it to be fixed soon.

Then again, you did make me think. Perhaps the solution should be to offer both. I was thinking about a MapControlGL version, but in that case we would need maximal code reuse. A better way maybe to add a method to switch render mode, like we have in the WPF MapControl, switching between Xaml and Skia. This way we can just go forward with both implementations. For the user the switch will be very easy.

**Comment by pauldendulk** [comment]
And we should indeed submit an issue. Preferably with a minimal reproducible sample. But I have two issues open with no response so far. https://github.com/mono/SkiaSharp/issues?q=is%3Aissue+is%3Aopen+pauldendulk

**Comment by pauldendulk** [comment]
Perhaps we should also start to dig into SkiaSharp more. Get some knowledge of the code. I cloned it only once, many years ago. Perhaps we could contact other users, like the people building Avalonia.

**Comment by mattleibow** [comment]
This might be a duplicate of #1121

**Comment by charlenni** [comment]
The above comment belongs to #865.

**Comment by pauldendulk** [comment]
@mattleibow This report is for Android. I am not completely sure there is no flaw in Mapsui here but when I go back to non-GL there is no crash which does indicate it is a SkiaSharp problem.

**Comment by pauldendulk** [comment]
@mattleibow Tested with 1.68.2-pr.1180.4 and could not reproduce the problem as reported above. Pushed it to this branch: https://github.com/Mapsui/Mapsui/tree/research/skiasharp-1.68.2-pr.1180.4.

It took a bit longer because 1.68.2-pr.1180.4 is not considered the latest version. I ended up replacing it manually in the csproj. Also we are busy getting the children's homeschooling program up and running. They are 7 and 9 and suddenly have to work on laptops and have a lot of questions :)
